File: mvc/src/main/java/me/kzv/simpleboardmvc/web/entity/Reply.java
package me.kzv.simpleboardmvc.web.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Reply extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reply_id")
    private Long id;
    private String text;
    private Boolean isParent; // 부모 댓글인지 아닌지
    private Long groupId; // isParent 가 false 인 경우 자식 댓글이므로 그룹 아이디가 있어야 한다. - nullable 부모 댓글일 경우 null
    private Boolean isActive;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="member_id")
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private Post post;

    @Builder
    protected Reply(String text, Member member, Boolean isParent, Post post, Long groupId) {
        this.text = text;
        this.member = member;
        this.isParent = isParent;
        this.post = post;

        this.groupId = groupId;
        this.isActive = true;
    }
}

File: backend/src/main/kotlin/me/kzv/simpleboard/domain/reply/entity/Reply.kt
package me.kzv.simpleboard.domain.reply.entity

import jakarta.persistence.*
import me.kzv.simpleboard.domain.BaseEntity
import me.kzv.simpleboard.domain.board.entity.Board
import me.kzv.simpleboard.domain.member.entity.Member

@Entity
class Reply(
    content: String,
    board: Board,
    replier: Member,
    parenId: Long?
) : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String = content
        private set

    val parentId: Long? = parenId

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_id")
    val board: Board = board

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    val replier: Member = replier
}